models/test-data-models.js

```diff
@@ -19,23 +19,37 @@ exports.selectArticleById = (article_id) => {
   });
 };
 
-exports.selectArticles = () => {
-  const query = `SELECT articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, 
+exports.selectArticles = (topic) => {
+  let query = `SELECT articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, 
   COUNT(comments.article_id) AS comment_count 
   FROM articles LEFT JOIN comments 
-  ON articles.article_id = comments.article_id 
-  GROUP BY
-  articles.title,
-  articles.topic,
-  articles.author,
-  articles.created_at,
-  articles.votes,
-  articles.article_id
-  ORDER BY
-  created_at DESC;`;
-  return db.query(query).then(({rows}) => {
-    return rows
-  })
+  ON articles.article_id = comments.article_id`;
+
+  const queryParams = [];
+
+  if (topic) {
+    query += ` WHERE articles.topic = $1`;
+    queryParams.push(topic);
+  }
+
+  query += ` GROUP BY
+    articles.title,
+    articles.topic,
```

Comment:
The `GROUP BY` only needs to group _article_id_ because they are the unique identifiers for each article